### ISSUE TITLE
Unable to provision ESP32-S3/C6 with Google Home on iOS (CON-1353)

### ISSUE THREAD

**Issue by ghost** [espressif/esp-matter#1097]
**Describe the bug**
When trying to add either an ESP32-C6 or ESP32-S3 as a Matter device in Google Home, I receive an error stating: "Unable to Add Accessory. Home couldn't connect to this accessory." The device successfully connects when provisioned with HomeKit on iOS, but not with Google Home on iOS. However, it works correctly when provisioned on Google Home using an Android device.

**Environment**
- ESP-Matter Commit Id:
Using esp_matter component version 1.3.0 which corresponds to this [commit](https://github.com/espressif/esp-matter/commit/d76d4857d768b32eb215e825448d938018e5691a)
- ESP-IDF Commit Id:
- ESP-IDF v5.2.2
- SoC (eg: ESP32 or ESP32-C3):
ESP32-C6 and ESP32-S3
- Device Logs (Please attach the log file):
[logs-esp32s3.txt](https://github.com/user-attachments/files/17168752/logs-esp32s3.txt)

- Host Machine OS:
Linux
- Host Machine Python version:
3.12
- Commissioner app and versions if present:
- Home hub app and versions if present:
Google Home 3.23.102
- Commissioner's logs if present:

**Any additional details**
- I'm using an iPhone 15 Pro Max running on iOS 18.0, and I also tested on iOS 17.6.1.
- I tested on two separate Wi-Fi networks and tried both Matter over OpenThread and Matter over Wi-Fi, but the issue persisted in all scenarios.
- Same issue as this : https://github.com/espressif/esp-matter/issues/987

**Comment by sayondeep** [comment]
@AntoineSX 
I was able to recreate the issue. We are looking into it

**Comment by NomNom1992** [comment]
@sayondeep Hello, Im looking for this problem too. I try connect matter device by google home on IOS, and got same error. I tried that with a Tapo matter socket, running on a Realtek chip. It runs normally with the same test.

**Comment by mgoeller** [comment]
Can we solve this please @sayondeep? I am experiencing this as well, and it is affecting the reputation of our devices sold! Please expedite this issue!

**Comment by 9zigen** [comment]
I have the same issue.

**Comment by jzwyssig** [comment]
We are also experiencing this issue and it is significantly impacting our development process. A prompt resolution would be greatly appreciated. Please let us know if there's any additional information or testing needed from our side to assist in addressing this.

**Comment by VaishaliAvhale** [comment]
I tried commissioning the ESP-Matter latest Thread and Wi-Fi light using ESP32-C6, and it was successfully commissioned and added to the iOS Google Home App.

Initially, when I attempted to commission the Thread light using the ESP32-C6, I encountered the same issue you mentioned. However, after retrying, it worked successfully.

Here are the details of my setup:

Google Home App Version: 3.26.103
Google Nest Hub (2nd Generation):
Fuchsia Version: 22.20240805.103.161
Software Version: 62.0.7.676990136

**Comment by shubhamdp** [comment]
If you’d like to maintain a degree of rudeness while still being professional, here’s an alternative that conveys your frustration but keeps it reasonably composed:

@thalesfsp Is this comment meant to be helpful, or are you just venting? What @VaishaliAvhale clearly mentioned is that she couldn’t reproduce the issue and has shared her setup details.

If you actually want this issue resolved, it would be more productive to provide useful information—such as reproduction steps, setup details, or logs—rather than speculative remarks.

**Comment by jzwyssig** [comment]
@VaishaliAvhale I experience the same problem with an ESP32-C6, we use the official docker container: esp-matter:release-v1.2_idf_v5.2.1.

- Commissioning works on Google Home with a Google Pixel Device
- Commissioning works on Apple Home with iOS 18.3.2
- Commissioning works if I share the pairing from Apple Home with Google Home on iOS
- Commissioning **FAILS** if I try directly Google Home on iOS 18.3.

**Comment by VaishaliAvhale** [comment]
Hi @jzwyssig,

I tried the same thing on my end but was unable to reproduce the issue. Could you please share the device logs and also let me know what you see on iOS Google Home when the commissioning fails?

For reference, I used the following specifications:

ESP-Matter Light: v1.2
Nest Hub 2nd Generation: 63.84.9.71343048
iPadOS: 18.4
Google Home App: 3.30.102

**Comment by jzwyssig** [comment]
Hi @VaishaliAvhale 

Thank you for testing. I tried the same with the **example/light** from the latest commit of the **esp-matter release/v1.2** repository. As described, I used the Docker container for **release v1.2** and **ESP-IDF v5.2.1**. I compiled the using the following commands:
```sh
idf.py set-target esp32c6
idf.py build flash monitor
```

I scanned the QR code from the [documentation](https://docs.espressif.com/projects/esp-matter/en/latest/esp32c6/_images/matter_qrcode_20202021_3840.png) and started the pairing using Google Home on iOS (same as your version: 3.30.102).

The **same error occurs**, see the image below. I have attached the log for reference. 
I think the `I (61887) chip[DL]: BLE GAP connection terminated (con 1 reason 0x213)` could be an issue. However, as I already mentioned, the pairing works perfectly fine with chip-tool, Apple Home and Google Home on Android. Only Google Home on iOS is an issue.

### Additional Attempts:
- Tried with **iOS 18.4**  
- Used another Docker container with **Matter 1.3**  
- Issue **remains the same**  

Do you have any ideas on how to resolve this issue?

[esp32-c6.log](https://github.com/user-attachments/files/19267834/esp32-c6.log)

![Image](https://github.com/user-attachments/assets/ac4cdfe4-a988-481e-b5e9-4084c1a7af72)

**Comment by shubhamdp** [comment]
@jzwyssig These logs shows the successful commissioning attempt.

```
I (61757) chip[DIS]: CHIP minimal mDNS configured as 'Operational device'; instance name: 00AEDD0B9420805C-00000000A7F074B1.
...
I (61867) app_main: Commissioning complete
I (61867) esp_matter_core: Commissioning Complete
I (61867) app_main: Commissioning window closed
I (61857) chip[DL]: Write request/command received for CHIPoBLE TX CCCD characteristic (con 1 ) indicate = 0
I (61887) chip[DL]: CHIPoBLE unsubscribe received
I (61887) chip[DL]: BLE GAP connection terminated (con 1 reason 0x213)
```

After first commissioning (it would have added the device to iOS keychain fabric) it is expected that it releases BLE connection. Then it should open the commissioning window again so that it can add the device to Google Home's fabric.

Do you have logs after this where theres a `I (1897) app_main: Commissioning window opened` log after first commissioning is successful?

**Comment by jzwyssig** [comment]
@shubhamdp exactly, the Apple KeyChain fabric is paired successfully. But after that the Bluetooth connection fails. I provided the full log and as you can see there is no second commissioning attempt.

I guess the reason is the Bluetooth connection that fails to shut down. Do you have any other ideas?

**Comment by shubhamdp** [comment]
@jzwyssig for second commissioning bluetooth is not required, it uses the ip transport. Those failed logs are okay, it's just di-initializing already initialized bluetooth.

In other cases, once the first commissioning is complete, that administrator opens the commissioning window so that the app (here Google Home iOS app) can commission device in that fabric. But I do not see any logs showing that behaviour. And since this works on other commissioners, not sure how to debug this further.

Can you do one thing though, if we somehow get the logs from the mobile app may be we can see what commissioner is doing. Also, during commissioning, can you check if device is broadcasting its mdns service?

You would see `I (55337) chip[DIS]: Advertise operational node 00AEDD0B9420805C-00000000A7F074B1` log on the device. On host (linux/mac/any-other) you can start the discovery and see if that name pops up.

```
# linux
avahi-browse -d local _matter._tcp --resolve

# mac
dns-sd -B _matter._tcp
```

Please check this troubleshooting guide: https://github.com/project-chip/connectedhomeip/blob/master/docs/tips_and_troubleshooting/discovery_from_a_host_computer.md
